src/components/DiscordImage.tsx

```diff
@@ -0,0 +1,24 @@
+"use client";
+
+import Image from "next/image";
+import useFocus from "~/hooks/use-focus";
+
+type DiscordImageProps = React.ComponentProps<typeof Image>;
+
+export default function DiscordImage(props: DiscordImageProps) {
+  const focused = useFocus();
+
+  const imageSrc =
+    typeof props.src === "string" &&
+    // ignore placeholders
+    !props.src.includes("/embed") &&
+    props.src.includes(".gif")
+      ? props.src.replace(
+          focused ? ".webp" : ".gif",
+          focused ? ".gif" : ".webp"
+        )
+      : props.src;
```

Comment:
```suggestion
  const imageSrc =
    typeof src === "string" &&
    // ignore placeholders
    !src.includes("/embed") &&
    src.includes(".gif")
      ? src.replace(
          focused ? ".webp" : ".gif",
          focused ? ".gif" : ".webp"
        )
      : src;
```
Won't be preperly formatted but not a problem, I'll handle that